reframe/listener.go

```diff
@@ -289,12 +291,14 @@ func (listener *ReframeListener) Provide(ctx context.Context, pr *client.Provide
 				//     * when currentChunk disappears between restarts as it doesn't get persisted until it's advertised
 				//     * when the same cid comes multiple times within the lifespan of the same chunk
 				//	   * after a error to generate a replacement chunk
-				err := listener.chunker.addCidToCurrentChunk(ctx, c, func(cc *cidsChunk) error {
-					return listener.notifyPutAndPersist(ctx, cc)
-				})
-				if err != nil {
-					log.Errorw("Error adding a cid to the current chunk. Continuing.", "cid", c, "err", err)
-					continue
+				if node.chunk == nil {
```

Comment:
That was the main bug that has led to the same cids been readvertised over and over again.